brightray/browser/url_request_context_getter.cc

```diff
@@ -216,29 +217,46 @@ net::URLRequestContext* URLRequestContextGetter::GetURLRequestContext() {
 
     storage_->set_network_delegate(delegate_->CreateNetworkDelegate());
 
+    std::unique_ptr<net::CookieStore> cookie_store;
+    scoped_refptr<net::SQLiteChannelIDStore> channel_id_db;
+    // Create a single task runner to use with the CookieStore and
+    // ChannelIDStore.
+    scoped_refptr<base::SequencedTaskRunner> cookie_background_task_runner =
+        base::CreateSequencedTaskRunnerWithTraits(
+            {base::MayBlock(), base::TaskPriority::BACKGROUND,
+             base::TaskShutdownBehavior::BLOCK_SHUTDOWN});
     auto cookie_path = in_memory_
                            ? base::FilePath()
                            : base_path_.Append(FILE_PATH_LITERAL("Cookies"));
-    std::unique_ptr<net::CookieStore> cookie_store = content::CreateCookieStore(
-        content::CookieStoreConfig(cookie_path, false, false, nullptr));
-    storage_->set_cookie_store(std::move(cookie_store));
+    if (!in_memory_) {
+      channel_id_db = new net::SQLiteChannelIDStore(
+          base_path_.Append(FILE_PATH_LITERAL("Origin Bound Certs")),
+          cookie_background_task_runner);
+    }
+    std::unique_ptr<net::ChannelIDService> channel_id_service(
+        new net::ChannelIDService(
+            new net::DefaultChannelIDStore(channel_id_db.get())));
+    content::CookieStoreConfig cookie_config(cookie_path, false, false,
+                                             nullptr);
+    cookie_config.channel_id_service = channel_id_service.get();
+    cookie_config.background_task_runner = cookie_background_task_runner;
+    cookie_store = content::CreateCookieStore(cookie_config);
+    cookie_store->SetChannelIDServiceID(channel_id_service->GetUniqueID());
```

Comment:
I am not sure what is crbug.com/548423 about since its restricted to chromium developers. But the original cause for DCHECK is because of these checks in https://cs.chromium.org/chromium/src/net/url_request/url_request_http_job.cc?l=193-228 and they are all based on unique id from `ChannelIDService`, so if that api is going away then it should also be refactoring that logic and eventually when its done we will align with upstream's new api.